scripts/ci/mini-lz4.js

```diff
@@ -0,0 +1,115 @@
+function uncompress(source, uncompressedSize) {
+/*
+based off https://github.com/emscripten-core/emscripten/blob/main/third_party/mini-lz4.js
+====
+MiniLZ4: Minimal LZ4 block decoding and encoding.
+
+based off of node-lz4, https://github.com/pierrec/node-lz4
+
+====
+Copyright (c) 2012 Pierre Curto
```

Comment:
Note that the copyright notice will remain in ``soljson.js`` the way this is currently set up. I moved it into the function to clarify its scope - but not sure what's the general policy for stuff like this with js code... I guess usually JS code is minified in the end, loosing license information...

Reply:
I tried removing all comments and it reduced it from 4 to 2.5 kB so in relative terms they add a lot. But the whole binary is 8 MB so I don't think this matters much in the grand scheme of things. From a legal standpoint it's probably safest to keep it like this.

And overall, I'm against minifying this code. I'd actually unminify the other part because it would give us better stack traces and I had to actually dive into this part more than once and it would have been convenient it was more readable.

Reply:
My point is rather the opposite: it may be confusing to have licenses in ``soljson.js``, which only apply to some parts - and I suppose minification counts as compilation in js land, so removing them via minification I guess might be fine :-). We could of course minify and then beautify again, but that's rather silly :-). Not sure what's actually the valid and reasonable thing to do for sth like this. Maybe a link only would be fine, but idk. I'll keep them in here for now.

Reply:
Added a comment about the scope of the license for now, but I'm open to doing it differently...
In my understanding of ``The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.`` it should be in there, but I guess things distributed as in javascript are usually fine with violating this, so idk... IANAL

Reply:
I wish licensing was a simple as just slapping on an SPDX identifier. Having license copypasta everywhere is seriously annoying

Reply:
We should just store all the licenses on IPFS and start pasting hashes of the content :)

Reply:
There is the licenses template which could be extended to include these.

Reply:
@axic Do you mean in addition to the comments in the js part or instead of them?
But yeah - we could probably even conditionally add it there only for emscripten builds...
Technically, it might be sane to keep the license part in ``mini-lz4.js`` itself, but remove the license comment when copying it to ``soljson.js``, while having it in ``cmake/templates/license.h.in``, s.t. the ``solidity_license()`` export reports it...

Reply:
What I'd do:
1. Keep comments in the sources as is. It helps tracking where something is from.
2. Include the new license tags with file links into the template, as we do for every imported piece.
3. Trim the comments from the JS files during final assembly. The js/wasm file shouldn't have any of that as comments.

Reply:
Yeah, that's pretty much what I meant.
Would you bother to include the license into ``cmake/templates/license.h.in`` conditionally only for the emscripten builds or just unconditionally include it (so also for native ``solc`` binaries, even though it's not actually in those)?`
It shouldn't be too hard to only include it if the cmake ``EMSCRIPTEN`` flag is true...

Reply:
@axic Does the rest of the PR look sane to you :-)?

Reply:
Reading the discussion about "Public Domain" in the json-cpp part of the licensing template, I'm wondering about whether the base64 snippet is an issue (which is "Public Domain" as of https://developer.mozilla.org/en-US/docs/MDN/About#copyrights_and_licenses)...

Reply:
Actually ``evmc`` is also listed in there unconditionally, even though it's not part of the ``solc`` binary...